SubmittingPatches.rst

```diff
@@ -62,468 +66,207 @@ then you just add a line saying ::
-
-   For example, you should set "Backport: jewel, kraken" in your Redmine ticket
-   to indicate that you are fixing a bug that exists on the "jewel" and
-   "kraken" branches and that you desire that your change be cherry-picked to
-   those branches after it is merged into master.
-
-Q: How to include ``Reviewed-by: tag(s)`` in my pull request?
-
-A: You don't. If someone reviews your pull request, they should indicate they
-   have done so by commenting on it with "+1", "looks good to me", "LGTM",
-   and/or the entire "Reviewed-by: ..." line with their name and email address.
-
-   The developer merging the pull request should note positive reviews and
-   include the appropriate Reviewed-by: lines in the merge commit.
+Title of commit
+---------------
 
+The text up to the first empty line in a commit message is the commit
+title. It should be a single short line of at most 72 characters,
+summarizing the change, and prefixed with the
+subsystem or module you are changing. For example::
 
-2. Patch submission via ceph-devel@vger.kernel.org
---------------------------------------------------
+     mds: add perf counter for finisher of MDSRank
 
-The best way to generate a patch for manual submission is to work from
-a Git checkout of the Ceph source code. You can then generate patches
-with the 'git format-patch' command. For example,
+More examples can be obtained from the git history of the master branch::
 
-.. code-block:: bash
+     git log
 
-   $ git format-patch HEAD^^ -o mything
+It is also conventional to use the imperative mood in the commit title. For
+example::
 
-will take the last two commits and generate patches in the mything/
-directory. The commit you specify on the command line is the
-'upstream' commit that you are diffing against. Note that it does
-not necessarily have to be an ancestor of your current commit. You
-can do something like
+     osd: make the ClassHandler::mutex private
 
-.. code-block:: bash
 
-   $ git checkout -b mything
-   # ... do lots of stuff ...
-   $ git fetch
-   # ...find out that origin/unstable has also moved forward...
-   $ git format-patch origin/unstable -o mything
+PR best practices
+-----------------
 
-and the patches will be against origin/unstable.
+PRs should be opened on branches contained in your fork of
+https://github.com/ceph/ceph.git - do not push branches directly to
+``ceph/ceph.git``.
 
-The ``-o`` dir is optional; if left off, the patch(es) will appear in
-the current directory. This can quickly get messy.
+PRs should target "master". If you need to add a patch to a stable branch, such
+as "nautilus", see the file ``SubmittingPatches-backports.rst``.
 
-You can also add ``--cover-letter`` and get a '0000' patch in the
-mything/ directory. That can be updated to include any overview
-stuff for a multipart patch series. If it's a single patch, don't
-bother.
+The `Title of commit`_ explains that commit message
+titles should look like this::
```

Comment:
Okay but how should they title the PR? What if they have multiple commits? Then the PR title is usually empty (IIRC) when github opens the PR creation page. We should suggest using the title from the most relevant commit. It is also okay to come up with a completely new title as long as it conforms to the commit title conventions.

Reply:
We should also address what the PR body should include. I generally don't spend too much effort on the PR body as most information should be in the commit messages instead. PR body should mostly be used as a center for information about the PR and notices to the maintainers?

Reply:
agreed. i think PR body is a place for sending messages to reviewer and whoever are interested to the PR. in some cases, we use it as a notepad to 

- document the todo list.
- the performed tests , the test not yet performed.
- the performance benchmarks 
- links for further reading / related PRs

i don't want to codify it. just like the title of an article, the title of a PR should enable its reader to make a decision right away if he/she should take a closer look with a glimpse .